connectors/sources/servicenow.py

```diff
@@ -327,6 +434,111 @@ async def ping(self):
             logger.exception("Error while connecting to the ServiceNow.")
             raise
 
+    def _format_doc(self, data):
+        """Format document for handling empty values & type casting.
+
+        Args:
+            data (dict): Fetched record from ServiceNow.
+
+        Returns:
+            dict: Formatted document.
+        """
+
+        data = {key: value for key, value in data.items() if value}
+        data.update(
+            {
+                "_id": data["sys_id"],
+                "_timestamp": iso_utc(parser.parse(data["sys_updated_on"])),
+            }
+        )
+        return self.serialize(doc=data)
+
+    async def _attachment_fetcher(self, table_sys_id):
+        """Method to add attachment metadata to queue
+
+        Args:
+            table_sys_id (str): Table Id for fetching attachment metadata.
+        """
+
+        async for attachment_metadata in self.servicenow_client.fetch_data(
```

Comment:
suggestion: I think we should have more high-level methods in the `ServiceNowClient` like `fetch_attachments` or `get_attachments` (see: [MySQLClient](https://github.com/elastic/connectors-python/blob/main/connectors/sources/mysql.py#L204)). Makes it easier to read and easier to use.